sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -62,10 +62,14 @@ class Iso8601TimestampFormatter(
       val parsed = formatter.parse(s)
       val parsedZoneId = parsed.query(TemporalQueries.zone())
       val timeZoneId = if (parsedZoneId == null) zoneId else parsedZoneId
-      val zonedDateTime = toZonedDateTime(parsed, timeZoneId)
+      // Parsed input might not have time related part. In that case, time component is set to
+      // zeros.
+      val parsedLocalTime = parsed.query(TemporalQueries.localTime)
+      val localTime = if (parsedLocalTime == null) LocalTime.MIDNIGHT else parsedLocalTime
+      val localDate = getLocalDate(s, parsed)
+      val zonedDateTime = ZonedDateTime.of(localDate, localTime, timeZoneId)
```

Comment:
@cloud-fan, did we pull these codes out of `toZonedDateTime`? I thought we can just fix it in `toZonedDateTime`.